src/panel/direction/DirectionPanel.jsx

```diff
@@ -157,6 +177,25 @@ export default class DirectionPanel extends React.Component {
         fire('set_destination', destination);
       }
     }
+
+    let poiInfo;
```

Comment:
By putting the reverse geocoding here, you postpone it to after the route is computed and displayed, whereas it could be done in parallel.
The two geocoding themselves could be done in parallel, but using two `await` make it in practice synchronized.
As a result, it's slower than needed.

This is also a bit strange to find these operations in a method called `computeRoutes` which explicitly tries to limit its scope to a single purpose. IMO it should be done in the `changeDirectionPoint`, so everything can be decoupled and can run more effectively.

Reply:
Hi,
I agree both requests must be made in parallel, I'm trying to fix it.
Regarding the place where the code should be, I initially chose computeRoutes because it seemed to be the only function called on field completion and on page reload, which is not the case for changeDirectionPoint (yet).
Maybe changeDirectionPoint only should compute addresses, and computeRoutes should call changeDirectionPoint, and if it does that, it could stop doing setStates directly... 🤔 
So I'm moving the code around hoping to find something more elegant, but I may need your help to confirm what is the nicest place/way to do it.

Reply:
Another side effect of calling Idunn here, is that new requests are executed on each compute (eg when another vehicle is selected). That should not be necessary.